web3/_utils/async_tools.py

```diff
@@ -0,0 +1,37 @@
+from functools import singledispatch, wraps
+import asyncio
+import inspect
+import types
+from typing import Any, Callable, Generator
+
+
+def _is_awaitable(co: Generator[Any, None, Any]) -> bool:
```

Comment:
Since this is a direct passthrough I think you can drop the utility function and just directly use `inspect.isawaitable`